carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -259,10 +259,10 @@ void WMBroadcaster::addRegionAccessRule(std::shared_ptr<Geofence> gf_ptr, const
   gf_ptr->regulatory_element_ = regulatory_element;
 }
 
-void WMBroadcaster::addRegionMinimumGap(std::shared_ptr<Geofence> gf_ptr, double min_gap, const std::vector<lanelet::Lanelet>& affected_llts, const std::vector<lanelet::Area>& affected_areas) const
+void WMBroadcaster::addRegionMinimumGap(std::shared_ptr<Geofence> gf_ptr,  const cav_msgs::TrafficControlMessageV01& msg_v01, double min_gap, const std::vector<lanelet::Lanelet>& affected_llts, const std::vector<lanelet::Area>& affected_areas) const
 {
   auto regulatory_element = std::make_shared<lanelet::DigitalMinimumGap>(lanelet::DigitalMinimumGap::buildData(lanelet::utils::getId(), 
-                                        min_gap, affected_llts, affected_areas, { lanelet::Participants::VehicleCar }));
+                                        min_gap, affected_llts, affected_areas, invertParticipants(participantsChecker(msg_v01))));
```

Comment:
Looking over the implementation, I don't think you need to invert the provided participants. The message specified which participants are restricted and your implementation of DigitalMinimumGap works the same way so I think a call to participantChecker() is sufficient.

Reply:
https://github.com/usdot-fhwa-stol/carma-platform/pull/1197/commits/b61c21edc31aa0b79c0e2e72ad8ddcd2f182faa8